distributed/__init__.py

```diff
@@ -46,7 +47,25 @@
 from .worker import Reschedule, Worker, get_client, get_worker, print, secede, warn
 from .worker_client import local_client, worker_client
 
-versions = get_versions()
-__version__ = versions["version"]
-__git_revision__ = versions["full-revisionid"]
-del get_versions, versions
+
+def __getattr__(name):
+    global __version__, __git_revision__
+
+    if name == "__version__":
+        import sys
+
+        if sys.version_info >= (3, 8):
+            from importlib.metadata import version
+        else:
+            from importlib_metadata import version
+
+        __version__ = version("distributed")
+        return __version__
+
+    if name == "__git_revision__":
+        from ._versioneer import get_versions
+
+        __git_revision__ = get_versions()["full-revisionid"]
```

Comment:
If this is expensive, we might want to cache it?

Reply:
ah, right. I'm not used to the module level getattr stuff. spooky

Reply:
based on this: https://github.com/python/cpython/blob/a481be0c5f5089ed5e81b9855f71eae12039cad2/Lib/concurrent/futures/__init__.py#L40-L53